frontend/src/pages/AddAccessCode.tsx

```diff
@@ -0,0 +1,93 @@
+import { ChevronLeftIcon } from '@heroicons/react/24/outline';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import i18n from '../common/config/i18n';
+import AccessCodeForm, { AccessCodeFormTypes } from '../components/AccessCodeForm';
+import Button from '../components/Button';
+import CardBody from '../components/CardBody';
+import CardHeader from '../components/CardHeader';
+import Card from '../layouts/CardLayout';
+import PageLayout from '../layouts/PageLayout';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { REGEX } from '../common/constants/patterns';
+import { useForm } from 'react-hook-form';
+import { useCreateAccessCodesMutation } from '../services/organization/organization.service';
+import { useErrorToast } from '../hooks/useToast';
+import { InternalErrors } from '../common/errors/internal-errors.class';
+import LoadingContent from '../components/LoadingContent';
+
+const validationSchema = yup.object({
+  code: yup
+    .string()
+    .required(`${i18n.t('general:validation.required')}`)
+    .min(2, `${i18n.t('access_codes:validation.min')}`)
+    .max(10, `${i18n.t('access_codes:validation.max')}`)
+    .matches(REGEX.NAME_REGEX, `${i18n.t('general:validation:pattern')}`),
+  startDate: yup.date().required(`${i18n.t('general:validation.required')}`),
+  endDate: yup.date().optional(),
+});
+
+const AddAccessCode = () => {
+  const navigate = useNavigate();
+
+  const { mutateAsync: createAccessCodeMutation, isLoading } = useCreateAccessCodesMutation();
+
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<AccessCodeFormTypes>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    resolver: yupResolver(validationSchema),
+  });
+
+  const onNavigateBack = () => {
+    navigate(-1);
```

Comment:
aici ar trebui sa specificam care este pagina la care se intoarce pentru ca in cazul in care sa da share la LINK nu o sa se stie.